keps/sig-cloud-provider/2699-add-webhook-hosting-to-ccm/README.md

```diff
@@ -0,0 +1,541 @@
+manager, err := cloudManagerBuilder(wait.NeverStop)
+if err != nil {
+klog.Fatalf("unable to construct cloud manager: %v", err)
+}
+err := command.start()
+```
+
+This will not alter the existing extension hooks in the controller manager 
+framework, as they are critical for backward compatibility. The builders
+are meant to be an abstraction layer on top to make the extensions easier to 
+use. So for the existing controller manager code you might see changes like:
+
+```
+cloudControllerManagerBuilder.registerController("nodeipamcontroller", handler)
+cloudControllerManagerBuilder.deregisterController("servicecontroller")
+```
+
+
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+
+No need to outline all of the test cases, just the general strategy. Anything
+that would count as tricky in the implementation, and anything particularly
+challenging to test, should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations). Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+There will be no e2e tests in K/K as we do not build a deployable CCM in K/K.
+There will be integration tests for the builder patterns.
+In addition we will rely on K/cloud-provider-gcp to demonstrate that the PVL mutating webhook comes up.
+
+### Graduation Criteria
+
+#### Alpha
+
+- Have the sample CCM come up and able to serve PVL mutating webhook.
+- Requires that we get a PVL mutating webhook written for at least 1 Cloud Provider.
+
+#### Beta
+
+- TBD
+
+#### GA
+
+- TBD
+
+**Note:** Generally we also wait at least two releases between beta and
+GA/stable, because there's no opportunity for user feedback, or even bug reports,
+in back-to-back releases.
+
+**For non-optional features moving to GA, the graduation criteria must include
+[conformance tests].**
+
+[conformance tests]: https://git.k8s.io/community/contributors/devel/sig-architecture/conformance-tests.md
+
+#### Deprecation
+
+- Not deprecated
+
+### Upgrade / Downgrade Strategy
+
+- Upgrade is not believed to be an issue at this point.
+- Currently we are leaving upgrade as an issue for the Cloud Provider
+
+### Version Skew Strategy
+
+- We are currently assuming that this will be deployed as part of the control
+plane. We assume it will be upgraded with the KAS, KCM and CCM.
+
+## Production Readiness Review Questionnaire
+
+- TBD
+
+### Feature Enablement and Rollback
+
+- TBD
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+This will be built into the CCM by the Cloud Provider. Code must be written
```

Comment:
As I recall, the CCM has featuregates.  This seems suitable for a featuregate so that consumers will be aware that the flags being set are considered alpha by the library providing them.

Reply:
Added for tracking purposes.